source/Camera/InfoUtilOther/ZividBenchmark/ZividBenchmark.cpp

```diff
@@ -0,0 +1,429 @@
+#include <Zivid/Zivid.h>
+#include <algorithm>
+#include <iostream>
+
+namespace
+{
+    using HighResClock = std::chrono::high_resolution_clock;
+    using Duration = std::chrono::nanoseconds;
+
+    Duration computeAverageDuration(const std::vector<Duration> durations)
```

Comment:
`const &` is what I had initially. However, CI complained when calling this function:
`printResultLine(names.at(i),                             formatDuration(computeMedianDuration(durations.at(i))),                             formatDuration(computeAverageDuration(durations.at(i))));`
The problem was that a variable has not been created but called as a function argument. Do you suggest that the correct way to be to create variables instead? For example:
`const auto a = computeMedianDuration(durations.at(i));
            const auto b = computeAverageDuration(durations.at(i));
            printResultLine(names.at(i), formatDuration(a), formatDuration(b));`

Reply:
@knzivid you made the same comment below.

Reply:
Try again and show me the code and the error if it is still the same. This sounds weird to me. You should not add a temporary variable to work around this.

Reply:
Ok, will do.
Is this a conflict of oppinion? Or is it different when we have a vector of Duration compared to just a Duration?
https://github.com/zivid/cpp-extra-samples/pull/83#discussion_r398371892

Reply:
`Duration` is trivial to copy, as easy to copy as copying the reference to it. `std::vector<Duration>` is an expensive copy.

The reasoning above (and @knzivid suggestion for `Duration`) requires knowledge about the internals of Duration.

Therefore, to keep it simple i recommend passing fundamental types (basically non-classes) by value and the reset (classes) by const references.

This means that I think Duration can be passed as `const &`.

Reply:
> 
> 
> Try again and show me the code and the error if it is still the same. This sounds weird to me. You should not add a temporary variable to work around this.

It seems that this issue was not due to the reference. I cannot reproduce it. Maybe it had to do with `const `for that input argument, I don't know.

I've made the suggested change now.

However, I may have introduced something that is not good practice. Please check lines 15-17 from the commit below. I wanted to make the input argument of `computeMedianDuration(const std::vector<Duration> &inputDurations)` a `const`, same as for the `computeAverageDuration`. However, since `std::sort` changes the vector, I had to copy the input argument inside the function. I suppose this is wrong and it is better to get rid of `const ` for this function?

https://github.com/zivid/cpp-extra-samples/pull/83/commits/bbbc08f4778bd2a579ac1e7e82c8c4b1d18b434f#diff-d8a114698146c40d02e597bce369da83R15

Reply:
Correct, that is bad.

If you actually are going to change the input, but not affect the caller, then you have a case where you pass by:

`computeMedianDuration(std::vector<Duration> inputDurations)`

Reply:
Done.